### PR TITLE
Ft create customer account 110060500

### PR DESCRIPTION
**Pull request by ChigboIO**
## Create Customer Account

Why?
When a user fills the signup form
His/her data should be stored in the database
And a welcome message containing the activation link should be sent to the
user's email address

How?
Generate user migration, model and controller
Write the script to process the submited form and store the valid data to
the database
Generate action mailer to send the user a welcome mail with the acivation
link

https://www.pivotaltracker.com/story/show/110060500
[finished #110060500]

### CODE REVIEW

**Comment by 0duaht** [comment]
I was able to continue the sign up even after the validation error was shown. Sign up button should be disabled if any of the front-end validations fail.

**Comment by oluosiname** [comment]
Tobi, I just noticed that as well. That was helpful, I'll get right to it

File: app/assets/javascripts/users.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		if ( $(this).val() === "")
+		{
+			$(this).addClass("invalid");
+			$(this).prev(".field-error").text("Field is required");
+			nameState = false;
+		}
+		else{
+			$(this).removeClass("invalid");
+			$(this).prev(".field-error").text("");
+			nameState = true;
```
Comment (makinwab) [inline app/assets/javascripts/users.js]: You are assigning value to a variable that was not declared or defined.

**Comment by ChigboIO** [comment]
@andela-bmakinwa 
Thank you for the comment
However, the file has been update to solve this, this is an old diff of the file. You can check the file in the `code view` to confirm

**Comment by 0duaht** [comment]
There's a slight bug with the view when model validation fails. For example, if email has already been taken.
Since the input fields retain their entries, button still stays disabled until all input fields are focused on. Even in cases where some input fields are without issues, the user still has to focus on them to meet the conditions for enabling the button.

**Comment by ChigboIO** [comment]
@andela-toduah 
Thank you for that observation
It has been fixed

**Pull request merged**